mobile/mysterium/entrypoint.go

```diff
@@ -622,15 +623,36 @@ func (mb *MobileNode) HealthCheck() *HealthCheckData {
 	}
 }
 
-// OrderStatusChangedCallback is a callback when order status changes.
-type OrderStatusChangedCallback interface {
-	OnChange(orderID uint64, status string)
+type OrderUpdatedCallbackPayload struct {
+	OrderID     int64
+	Status      string
+	PayAmount   float64
+	PayCurrency string
+}
+
+// OrderUpdatedCallback is a callback when order status changes.
+type OrderUpdatedCallback interface {
+	OnUpdate(payload *OrderUpdatedCallbackPayload)
 }
 
-// RegisterOrderStatusChangeCallback registers OrderStatusChanged callback.
-func (mb *MobileNode) RegisterOrderStatusChangeCallback(cb OrderStatusChangedCallback) {
-	_ = mb.eventBus.SubscribeAsync(pilvytis.AppTopicOrderStatusChanged, func(e pilvytis.AppEventOrderStatusChanged) {
-		cb.OnChange(e.ID, string(e.Status))
+// RegisterOrderUpdatedCallback registers OrderStatusChanged callback.
+func (mb *MobileNode) RegisterOrderUpdatedCallback(cb OrderUpdatedCallback) {
+	_ = mb.eventBus.SubscribeAsync(pilvytis.AppTopicOrderUpdated, func(e pilvytis.AppEventOrderUpdated) {
+		payload := OrderUpdatedCallbackPayload{}
+		id, err := shrinkUint64(e.ID)
+		if err != nil {
+			log.Err(err).Send()
```

Comment:
Maybe it's worth to also point out the level `Error()` and add some message so it's more clear whats going when reading logs.

Reply:
`log.Err(err)` is a shorthand for `log.Error().Err(err)`